lib/RecipeRunner.js

```diff
@@ -125,13 +125,28 @@ exports.RecipeRunner = {
       window.registerAction = registerAction;
     `;
 
-    const driver = new NormandyDriver(sandbox, extraContext);
+    const driver = new NormandyDriver(this, sandbox, extraContext);
     sandbox.sandboxedDriver = Cu.cloneInto(driver, sandbox, {cloneFunctions: true});
     sandbox.sandboxedRecipe = Cu.cloneInto(recipe, sandbox);
     sandbox.window = Cu.cloneInto({}, sandbox);
+    sandbox.navigator = Cu.cloneInto({}, sandbox);
 
     Cu.evalInSandbox(registerActionScript, sandbox);
     Cu.evalInSandbox(actionScript, sandbox);
     Log.debug('Finished running action');
   }),
+
+  heartbeatNotifications: [],
+
+  cleanup() {
+    if (this.heartbeatNotifications.length) {
+      this.heartbeatNotifications.forEach(heartbeat => {
+        let notice = heartbeat.notificationBox.getNotificationWithValue(`heartbeat-${heartbeat.options.flowId}`);
+        if (notice) {
+          heartbeat.notificationBox.removeNotification(notice);
+        }
+        this.heartbeatNotifications.shift();
```

Comment:
Is this safe to do while iterating over the list? I'm always a bit nervous about modifying the thing I'm iterating on.

Maybe just set the notification list to [] afterwards?